### PR TITLE
add Nvidia OptiX 3.x easyconfig

### PR DESCRIPTION
**Pull request by thiell**
OptiX library is a VMD requirement

### CODE REVIEW

File: easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+easyblock = 'Binary'
+name = 'OptiX'
+version = '3.8.0'
+
+homepage = 'https://developer.nvidia.com/optix'
+description = """OptiX is NVIDIA SDK for easy ray tracing performance.
+ It provides a simple framework for accessing the GPU’s massive ray tracing
+ power using state-of-the-art GPU algorithms."""
+
+toolchain = {'name': 'GNU', 'version': '4.9.3-2.25'}
```
Comment (boegel) [inline easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb]: @thiell why is this commented out?

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+]
+
+install_cmd = "./" + sources[0] + " --skip-license --prefix=%(installdir)s"
+
+sanity_check_paths = {
+    'files': ["include/optix.h", "include/optix_cuda.h", "lib64/liboptix.so", "lib64/liboptixu.so"],
+    'dirs': []
+}
+
+modextravars = { 'OPTIX_HOME': '%(installdir)s' }
```
Comment (boegel) [inline easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb]: please drop the spaces after `{` and before `}`

File: easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+##
+# This file is an EasyBuild reciPY as per https://github.com/hpcugent/easybuild
+# Authors::   Stephane Thiell <sthiell@stanford.edu>
+##
+easyblock = 'Binary'
+name = 'OptiX'
```
Comment (boegel) [inline easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb]: 'coding' style dictate empty line above `name = ...`

File: easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+version = '3.8.0'
+
+homepage = 'https://developer.nvidia.com/optix'
+description = """OptiX is NVIDIA SDK for easy ray tracing performance.
+ It provides a simple framework for accessing the GPU’s massive ray tracing
+ power using state-of-the-art GPU algorithms."""
+
+toolchain = {'name': 'GNU', 'version': '4.9.3-2.25'}
+
+#source_urls = ['http://developer.download.nvidia.com/compute/cuda/%(version_major_minor)s/Prod/local_installers/']
```
Comment (boegel) [inline easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb]: @thiell why is this commented out?

**Comment by boegel** [comment]
@thiell ping on tackling the remarks?

File: easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+toolchain = {'name': 'GNU', 'version': '4.9.3-2.25'}
+
+# Registration required. Download links:
+# https://developer.nvidia.com/designworks/optix/download
+# https://developer.nvidia.com/designworks/optix/downloads/legacy
+sources = ['NVIDIA-OptiX-SDK-%(version)s-linux64.run']
+
+checksums = [
+    'ecf63cf32b87defa36d3cd283d085314',    # NVIDIA-OptiX-SDK-3.8.0-linux64.run
+    '73eb689d6247e3d406cf3a38e80ec473',    # NVIDIA-OptiX-SDK-3.7.0-linux64.run
```
Comment (boegel) [inline easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb]: @thiell You should only have one checksum listed here, i.e. the one that corresponds to the version used here.

File: easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+]
+
+install_cmd = "./" + sources[0] + " --skip-license --prefix=%(installdir)s"
+
+sanity_check_paths = {
+    'files': ["include/optix.h",
+              "include/optix_cuda.h",
+              "lib64/liboptix.%s" % SHLIB_EXT,
+              "lib64/liboptixu.%s" % SHLIB_EXT],
```
Comment (boegel) [inline easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb]: collapse this to a single line?

File: easybuild/easyconfigs/o/OptiX/OptiX-3.9.0-GNU-4.9.3-2.25.eb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    '13b1953af766b82e0a4aaa28766a05a1',    # NVIDIA-OptiX-SDK-3.9.0-linux64.sh
+]
+
+install_cmd = "./" + sources[0] + " --skip-license --prefix=%(installdir)s"
+
+sanity_check_paths = {
+    'files': ["include/optix.h",
+              "include/optix_cuda.h",
+              "lib64/liboptix.%s" % SHLIB_EXT,
+              "lib64/liboptixu.%s" % SHLIB_EXT],
```
Comment (boegel) [inline easybuild/easyconfigs/o/OptiX/OptiX-3.9.0-GNU-4.9.3-2.25.eb]: single line?

Comment (thiell) [inline easybuild/easyconfigs/o/OptiX/OptiX-3.9.0-GNU-4.9.3-2.25.eb]: but then it's longer than 120 chars, so ok I will use only 2 lines

File: easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb (hunk 5 above)
Comment (thiell) [inline easybuild/easyconfigs/o/OptiX/OptiX-3.8.0-GNU-4.9.3-2.25.eb]: @boegel ok

File: easybuild/easyconfigs/o/OptiX/OptiX-3.9.0-GNU-4.9.3-2.25.eb (hunk 7 above)
Comment (boegel) [inline easybuild/easyconfigs/o/OptiX/OptiX-3.9.0-GNU-4.9.3-2.25.eb]: OK, then go with:

```python
'files': ['include/optix.h', 'include/optix_cuda.h',
          'lib64/liboptix.%s' % SHLIB_EXT, 'lib64/liboptixu.%s' % SHLIB_EXT],

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/573d96bb57c2a900948ee21654fbc221 for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2448.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/d9b43ecbc1de3e1621aaff0ab62192f8 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @thiell!

**Pull request merged**